### ISSUE TITLE
allow for removing very common words

### ISSUE THREAD

**Issue by McWashr** [jboynyc/textnets#56]
Howdy, I come from a background in implementing LDA topic modelling. I must say this is a really well-developed package that I was able to use really effectively. I love your work.

From what I can tell across documentation/in code, though, no way exists to specify that terms be filtered out if they are too frequent. This was quite a valuable feature when performing LDA, and I suspect it can help here when people are trying to explore the range of different topics. Often terms very common across large datasets contribute very little value / are noise.

I do not believe pre-processing of data is appropriate here, as that could potentially undermine the quality of the underlying data the model then works on. I am happy to look at making a pull request / seeing if I can implement this - figured I would first check if this is a kind of feature that you would be happy to see in this package.

Cheers

**Comment by jboynyc** [comment]
Hi @Jamal135, thanks for this suggestion as well as your kind words. Right now, if a word appears an equal number of times in every document, it will be disconnected from the main component, in which case setting `connected=True` will get rid of it. If a word appears in almost every document an even number of times, it will be connected with a weak edge that can likely be pruned by setting `remove_weak_edges=True`. 

However, I can see why what you describe would be useful. I can experiment with it, but I'm also happy to review a pull request if you want to dig into the code a bit.

**Comment by McWashr** [comment]
Over the next week or two, I will find time to dig in and see if I can find a good implementation. I'd agree 'connected' and 'remove_weak_edges' go some way to help. It will be interesting to see how results change with a more specific tool for removing very common words.

**Comment by jboynyc** [comment]
Closing this for now. This can already be achieved in an ad hoc using the `remove` keyword of `tokenized()` etc., so it seems low priority.

**Comment by McWashr** [comment]
Life kicked me in the butt, but I have returned. I implemented a max_docs logic onto the ``_matrix_from_tidy_text`` function, updated the ``Textnet`` class to support the input of that argument, and added relevant documentation in both docstring and the tutorial.

I am not familiar enough with this codebase to do full testing, though I was able to succeed with the Moon Landing example:

```python
import textnets as tn

tn.params["seed"] = 42

corpus = tn.Corpus(tn.examples.moon_landing)

t = tn.Textnet(corpus.tokenized(), min_docs=1, max_docs=1)

fig = t.plot(label_nodes=True,
       show_clusters=True,
       target='test.png')
```
![test](https://github.com/jboynyc/textnets/assets/50566368/483aba9b-4fe8-49cc-b7d8-547f536fc949)

I will place a pull request.
